examples/webserver/threads/src/main/java/io/helidon/examples/webserver/threads/ThreadService.java

```diff
@@ -0,0 +1,208 @@
+            // thread, plus provides us the ability to limit the number of concurrent computation requests
+            // we handle by limiting the thread pool work queue length (as defined in application.yaml)
+            Future<Double> future = platformExecutorService.submit(() -> compute(iterations));
+            response.send(future.get().toString());
+        } catch (RejectedExecutionException e) {
+            // Work queue is full! We reject the request
+            LOGGER.log(Level.WARNING, e);
+            response.status(Status.SERVICE_UNAVAILABLE_503).send("Server busy");
+        } catch (ExecutionException | InterruptedException e) {
+            LOGGER.log(Level.ERROR, e);
+            response.status(Status.INTERNAL_SERVER_ERROR_500).send();
+        }
+    }
+
+    /**
+     * Sleep for a specified number of seconds.
+     * The optional path parameter controls the number of seconds to sleep. Defaults to 1
+     *
+     * @param request  server request
+     * @param response server response
+     */
+    private void sleepHandler(ServerRequest request, ServerResponse response) {
+        int seconds = request.path().pathParameters().first("seconds").asInt().orElse(1);
+        response.send(String.valueOf(sleep(seconds)));
+    }
+
+    /**
+     * Fan out a number of remote requests in parallel.
+     * The optional path parameter controls the number of parallel requests to make.
+     *
+     * @param request  server request
+     * @param response server response
+     */
+    private void fanOutHandler(ServerRequest request, ServerResponse response) {
+        int count = request.path().pathParameters().first("count").asInt().orElse(1);
+        LOGGER.log(Level.INFO, "Fanning out " + count + " parallel requests");
+        // We simulate multiple client requests running in parallel by calling our sleep endpoint.
+        try {
+            // For this we use our virtual thread based executor. We submit the work and save the Futures
+            var futures = new ArrayList<Future<String>>();
+            for (int i = 0; i < count; i++) {
+                futures.add(virtualExecutorService.submit(() -> callRemote(rand.nextInt(5))));
+            }
+
+            // After work has been submitted we loop through the future and block getting the results.
+            // We aggregate the results in a list of Strings
+            var responses = new ArrayList<String>();
+            for (var future : futures) {
+                try {
+                    responses.add(future.get());
+                } catch (InterruptedException e) {
+                    responses.add(e.getMessage());
+                }
+            }
+
+            // All parallel calls are complete!
+            response.send(String.join(":", responses));
+        } catch (ExecutionException e) {
+            LOGGER.log(Level.ERROR, e);
+            response.status(Status.INTERNAL_SERVER_ERROR_500).send();
+        }
+    }
```

Comment:
Maybe mimicking structured concurrency constructs with existing tooling would better?

```java
        var futures = new ArrayList<CompletableFuture<String>>();
        try (var exec = Executors.newVirtualThreadPerTaskExecutor()) {
            for (int i = 0; i < count; i++) {
                futures.add(CompletableFuture.supplyAsync(() -> callRemote(rand.nextInt(5)), exec));
            }

            String result = futures.stream()
                    .map(CompletableFuture::join)
                    .collect(Collectors.joining(":"));
        }
```

Reply:
I intentionally chose to limit this to `Future` and direct use of `ExecutorServer.submit()` instead of introducing a large new API (`CompletableFuture` with its 70 methods) and indirect use of the `ExecutorService`. The existing code also parallels what is shown in `computeHandler` --  and I think that symmetry is important.